circleci/post.md

```diff
@@ -0,0 +1,288 @@
+4. Releasing the SDK
+
+Let's go through each one in details.
+
+## 1. Running Unit Tests, Integration Tests and Danger
+
+```yaml
+unit-tests:
+  macos:
+    xcode: "9.3.1"
+  working_directory: /Users/distiller/project
+  environment:
+    BUNDLE_PATH: vendor/bundle  # path to install gems and use for caching
+    ARTIFACTS_DIRECTORY: /Users/distiller/project/artifacts
+  steps:
+    - checkout
+    - restore_cache:
+        keys:
+        - gems-{{ checksum "Gemfile.lock" }}
+        # Fall back to using the latest cache if no exact match is found.
+        # - v1-gems-
+    - run: 
+        name: Create artifacts directory
+        command: mkdir $ARTIFACTS_DIRECTORY
+    # Install gems.
+    - run:
+        name: Bundle install
+        command: bundle check || bundle install
+        environment:
+          BUNDLE_JOBS: 4
+          BUNDLE_RETRY: 3
+    - save_cache:
+        key: gems-{{ checksum "Gemfile.lock" }}
+        paths:
+          - vendor/bundle
+    - run:
+        name: Pre-start simulator
+        command: xcrun instruments -w "iPhone 8 (11.3) [" || true
+    - run:
+        name: Run Instabug unit tests
+        command: set -o pipefail && xcodebuild -workspace Instabug/Instabug.xcworkspace -scheme AllUnitTests -sdk iphonesimulator -destination 'platform=iOS Simulator,OS=11.3,name=iPhone 8' -enableCodeCoverage YES test | xcpretty --color --report junit --output $ARTIFACTS_DIRECTORY/AllTests_unittest_results.xml
+    - run:
+        name: Run Danger
+        command: bundle exec Danger
+    - run: 
+        name: Run integration tests
+        command: set -o pipefail && xcodebuild -workspace Instabug/Instabug.xcworkspace -scheme InstabugIntegrationTests -sdk iphonesimulator -destination 'platform=iOS Simulator,OS=11.3,name=iPhone 8' -enableCodeCoverage YES test | xcpretty --color --report junit --output $ARTIFACTS_DIRECTORY/InstabugIntegrationTests_unittest_results.xml
+```
+
+First, we install all gems used by our project, like [Danger](http://danger.systems/) and [fastlane](https://fastlane.tools/) either by restoring the cache of a previous install, or by doing a fresh install.
+
+We then run our unit tests. Due to how our project is structured, we have around 15 framework targets, with a separate test target for each. To run all our unit tests at once, we have an `AllUnitTests` scheme that runs all test targets.
+
+After running our unit tests, we run Danger, which is great tool to automate some of the chores around code reviews. Here's what we use it for right now.
+
+1. Enforce having has a description and a link to a Jira issue for all pull requests
+2. Ensure that all pull requests that add new user-facing strings add localized versions of those strings.
+3. Run [xcov](https://github.com/nakiostudio/xcov) to generate a report about our tests coverage, and post it as a comment on the pull request
+4. Ensure that all pull requests that modify any UI file run UI tests before merging the pull request.
```

Comment:
Ensure that all pull requests that modify any UI file, run UI tests before merging the pull request.